### PR TITLE
Initial commit of SDL2 (2.0.4)

### PR DESCRIPTION
**Pull request by jimklimov**
Added a recipe for libsdl2 and friends, since sdl1.2.x is more or less considered obsolete by the project, and I'm looking at some programs that would need SDL2 to compile.
Package builds and publishes cleanly. Usability for the job - remains to be seen ;)

### CODE REVIEW

File: components/SDL2/Makefile [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+include $(WS_TOP)/make-rules/ips.mk
+
+PATH=/usr/gnu/bin:/usr/bin
+
+COMPONENT_PREP_ACTION = \
+			( cd $(@D) && $(CONFIG_SHELL) autogen.sh )
+
+CONFIGURE_OPTIONS +=	--sysconfdir=/etc
+CONFIGURE_OPTIONS +=	--libexecdir=$(CONFIGURE_LIBDIR.$(BITS))
+CONFIGURE_OPTIONS +=	--libdir=$(CONFIGURE_LIBDIR.$(BITS))
```
Comment (pyhalov) [inline components/SDL2/Makefile]: This should be automatic

File: components/SDL2/Makefile [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+
+#
+# Copyright (c) 2016 Jim Klimov
+#
+
+include ../../make-rules/shared-macros.mk
+
+COMPONENT_NAME=		SDL2
+COMPONENT_VERSION=	2.0.4
+COMPONENT_REVISION=	1
```
Comment (pyhalov) [inline components/SDL2/Makefile]: COMPONENT_REVISION is not necessary for the first revision

File: components/SDL2/sdl2.p5m [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+file path=usr/lib/$(MACH64)/libSDL2_test.a
+file path=usr/lib/$(MACH64)/libSDL2main.a
+file path=usr/lib/libSDL2-2.0.so.0.4.0
+link path=usr/lib/libSDL2-2.0.so.0 target=libSDL2-2.0.so.0.4.0
+link path=usr/lib/libSDL2-2.0.so target=libSDL2-2.0.so.0
+link path=usr/lib/libSDL2.so target=libSDL2-2.0.so
+file path=usr/lib/libSDL2.a
+file path=usr/lib/libSDL2_test.a
+file path=usr/lib/libSDL2main.a
+dir  path=usr/lib/cmake
```
Comment (pyhalov) [inline components/SDL2/sdl2.p5m]: We usually avoid delivering dir actions until it's  strictly necessary

Comment (jimklimov) [inline components/SDL2/sdl2.p5m]: and do we commonly have /usr/lib/cmake in a typical installation? which might not have cmake itself?

File: components/SDL2/Makefile (hunk 1 above)
Comment (jimklimov) [inline components/SDL2/Makefile]: Copypasted from SDL component...

File: components/SDL2/sdl2.p5m (hunk 3 above)
Comment (pyhalov) [inline components/SDL2/sdl2.p5m]: It doesn't matter, file action for usr/lib/cmake/SDL2/sdl2-config.cmake is enough

File: components/SDL2/Makefile [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+include $(WS_TOP)/make-rules/configure.mk
+include $(WS_TOP)/make-rules/ips.mk
+
+PATH=/usr/gnu/bin:/usr/bin
+
+COMPONENT_PREP_ACTION = \
+			( cd $(@D) && $(CONFIG_SHELL) autogen.sh )
+
+#CONFIGURE_OPTIONS +=	--sysconfdir=/etc
+#CONFIGURE_OPTIONS +=	--libexecdir=$(CONFIGURE_LIBDIR.$(BITS))
```
Comment (pyhalov) [inline components/SDL2/Makefile]: --libexecdir and --sysconfdir are not passed by default, but --libdir is added by default rules (see configure.mk)

File: components/SDL2/sdl2.p5m (hunk 3 above)
Comment (jimklimov) [inline components/SDL2/sdl2.p5m]: fixed

**Comment by jimklimov** [comment]
Hopefully addressed all the comments above (rebuilding and committing soon).

On a side note, do we have a good mechanism to use different HW-optimized features or builds as determined at runtime (like loopback mounts of libc.so back in the day)? (Anyhow a task for some other non-initial development and PR)

I ask because the SDL2 has a number of compile-time options for different CPU instructions that might make a difference on newer CPUs but might not work on older ones, but if I understand it correctly - it does not offer building a single binary that would pick correct implementation based on CPU capabilities, sadly...

**Comment by pyhalov** [comment]
No, we don't have such mechanism. 
Please, move files to components/library/SDL2

**Comment by jimklimov** [comment]
Is this good to go? IIRC it was a dependency for OpenTTD to be built... ;)

**Comment by pyhalov** [comment]
Yes, please, merge commits.

**Pull request merged**